### PR TITLE
chore: detect gestures on iOS

### LINKED ISSUE #1323: implement gesture detection - UIKit
The measure SDK should be able to detect 3 types of gesture

- Click
- Long click
- Scroll

These have custom event types and get saved to the DB once any gesture is detected.

### PR DESCRIPTION
**Pull request by adwinross**
# Description

Detect and save gestures.

## Related issue
Closes #1323

### CODE REVIEW

File: ios/MeasureSDK/MeasureSDK/Config/Config.swift [hunk 1]
```diff
@@ -33,5 +35,7 @@ struct Config: InternalConfig, MeasureConfig {
         self.eventsBatchingIntervalMs = 30000 // 30 seconds
         self.maxEventsInBatch = 500
         self.sessionEndThresholdMs = 60000 // 60 seconds
+        self.longPressTimeout = 0.5 // 0.5 seconds
```
Comment (adwinross) [inline ios/MeasureSDK/MeasureSDK/Config/Config.swift:38]: this is 500 ms as per [flutter documentation](https://api.flutter.dev/flutter/gestures/LongPressGestureRecognizer-class.html#:~:text=Recognizes%20when%20the%20user%20has,for%20500ms%20until%20it's%20recognized.).

File: ios/MeasureSDK/MeasureSDK/Config/InternalConfig.swift [hunk 2]
```diff
@@ -14,4 +14,11 @@ protocol InternalConfig {
 
     /// The threshold after which a session is considered ended. Defaults to 1 minute.
     var sessionEndThresholdMs: Number { get }
+
+    // TODO: check flutter for this info
```
Comment (adwinross) [inline ios/MeasureSDK/MeasureSDK/Config/InternalConfig.swift:18]: remove todo comment

File: ios/MeasureSDK/MeasureSDK/Config/Config.swift [hunk 3]
```diff
@@ -33,5 +35,7 @@ struct Config: InternalConfig, MeasureConfig {
...
+        self.scaledTouchSlop = 20 //
```
Comment (adwinross) [inline ios/MeasureSDK/MeasureSDK/Config/Config.swift:39]: The minimum threshold for scroll detection is 3.5 according to flutter's [dragStartDistanceMotionThreshold](https://api.flutter.dev/flutter/widgets/BouncingScrollPhysics/dragStartDistanceMotionThreshold.html) property.

File: ios/MeasureSDK/MeasureSDK/Config/Config.swift [hunk 4]
```diff
@@ -33,5 +35,7 @@ struct Config: InternalConfig, MeasureConfig {
         self.eventsBatchingIntervalMs = 30000 // 30 seconds
         self.maxEventsInBatch = 500
         self.sessionEndThresholdMs = 60000 // 60 seconds
+        self.longPressTimeout = 0.5 // 0.5 seconds
+        self.scaledTouchSlop = 3.5 // 3.5 points
```
Comment (adwinross) [inline ios/MeasureSDK/MeasureSDK/Config/Config.swift:39]: The minimum threshold for scroll detection is 3.5 according to flutter's [dragStartDistanceMotionThreshold](https://api.flutter.dev/flutter/widgets/BouncingScrollPhysics/dragStartDistanceMotionThreshold.html) property.

**Review by abhaysood — APPROVED** [review]

**Pull request merged**